File: njob-rest/src/main/java/com/github/nsorin/njobrest/model/Application.java
package com.github.nsorin.njobrest.model;

import java.util.List;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.Id;
import javax.persistence.ManyToOne;
import javax.persistence.OneToMany;
import javax.validation.constraints.Max;
import javax.validation.constraints.Min;
import javax.validation.constraints.NotNull;

import com.fasterxml.jackson.annotation.JsonBackReference;
import com.fasterxml.jackson.annotation.JsonManagedReference;

import org.hibernate.validator.constraints.URL;

import lombok.Getter;
import lombok.Setter;

@Entity
public class Application {

    @GeneratedValue
    @Id
    @Getter
    @Setter
    private Integer id;

    @Getter
    @Setter
    @NotNull
    private String name;

    @Getter
    @Setter
    @NotNull
    private String description;

    @Getter
    @Setter
    @NotNull
    @Min(1)
    @Max(5)
    private Integer priority;

    @Getter
    @Setter
    @URL
    private String offer;

    @JsonBackReference
    @ManyToOne
    @Getter
    @Setter
    private Company company;

    @JsonManagedReference
    @OneToMany(mappedBy = "application")
    @Getter
    @Setter
    private List<Document> documents;
}

File: njob-frontend/src/store/models/Application.js
import { Model } from "@vuex-orm/core"
import Company from "./Company"
import Document from "./Document"

export default class Application extends Model {

    static entity = 'application'

    static fields() {
        return {
            id: this.number(null),
            name: this.string(''),
            description: this.string(''),
            priority: this.number(1),
            company_id: this.number(null),
            company: this.belongsTo(Company, 'company_id'),
            documents: this.hasMany(Document, 'application_id')
        }
    }

}